submarine-server/server-core/src/main/java/org/apache/submarine/server/database/utils/MyBatisUtil.java

```diff
@@ -34,33 +34,39 @@
   private static final Logger LOG = LoggerFactory.getLogger(MyBatisUtil.class);
 
   private static SqlSessionFactory sqlSessionFactory;
+  private static SqlSessionFactory metastoreSqlSessionFactory;
 
   static {
+    SubmarineConfiguration conf = SubmarineConfiguration.getInstance();
+    sqlSessionFactory = buildSqlSessionFactory("mybatis-config.xml",
+        conf.getJdbcUrl(), conf.getJdbcUserName(), conf.getJdbcPassword());
+    metastoreSqlSessionFactory = buildSqlSessionFactory(
+        "mybatis-config-metastore.xml", conf.getMetastoreJdbcUrl(),
+        conf.getMetastoreJdbcUserName(), conf.getMetastoreJdbcPassword());
+  }
```

Comment:
@manirajv06 I guess that travis failure may be related to this.
Or where you have modified other database operations.

Reply:
I spent some more time on these issues again to get more details. There are 2 issues:

1. https://travis-ci.org/github/apache/submarine/jobs/705069335

==>  Preparing: INSERT INTO sys_dict(`id`, `dict_code`, `dict_name`, `description`, `create_by`, `create_time`, `update_time`) VALUES ( REPLACE(UUID(), '-', ''), ?, ?, ?, ?, now(), now()) 
==> Parameters: SYS_USER_SEX(String), name(String), description(String), null
[ERROR] 2020-07-06 02:03:29,612 method:org.apache.submarine.server.workbench.rest.SysDictRestApi.add(SysDictRestApi.java:107)
### Error updating database.  Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry 'SYS_USER_SEX' for key 'UK_SYS_DICT_DICT_CODE'
### The error may involve org.apache.submarine.server.workbench.database.mappers.SysDictMapper.insertSysDict-Inline
### The error occurred while setting parameters
### SQL: INSERT INTO sys_dict(`id`, `dict_code`, `dict_name`, `description`,                 `create_by`, `create_time`, `update_time`)     VALUES (               REPLACE(UUID(), '-', ''),             ?, ?, ?,                 ?, now(), now())
### Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry 'SYS_USER_SEX' for key 'UK_SYS_DICT_DICT_CODE'

I can able to reproduce the error in my master branch itself. SysUserRestApiTest extends CommonDataTest class and CommonDataTest#startTest (@BeforeClass) tries to load the data. docs/submarine-data.sql also populates the same data. Seems to be duplicate entry issue. (or) Am I missing anything?

2. https://travis-ci.org/github/apache/submarine/jobs/705069329

### Error querying database.  Cause: java.sql.SQLException: Error setting driver on UnpooledDataSource. Cause: java.lang.ClassNotFoundException: Cannot find class: com.mysql.jdbc.Driver
### The error may exist in org/apache/submarine/database/mappers/SysUserMapper.xml
### The error may involve org.apache.submarine.server.workbench.database.mappers.SysUserMapper.login
### The error occurred while executing a query
### Cause: java.sql.SQLException: Error setting driver on UnpooledDataSource. Cause: java.lang.ClassNotFoundException: Cannot find class: com.mysql.jdbc.Driver
org.apache.ibatis.exceptions.PersistenceException: 
### Error querying database.  Cause: java.sql.SQLException: Error setting driver on UnpooledDataSource. Cause: java.lang.ClassNotFoundException: Cannot find class: com.mysql.jdbc.Driver
### The error may exist in org/apache/submarine/database/mappers/SysUserMapper.xml
### The error may involve org.apache.submarine.server.workbench.database.mappers.SysUserMapper.login
### The error occurred while executing a query
### Cause: java.sql.SQLException: Error setting driver on UnpooledDataSource. Cause: java.lang.ClassNotFoundException: Cannot find class: com.mysql.jdbc.Driver

MySQL driver class is missing...Seems to be intermittent issue?

Reply:
@kevin85421 Can you help to see this problem?

Reply:
@kevin85421 Had a chance to look at this?

Reply:
@manirajv06 Thanks for the contribution.
I'm going to help you fix the test failure.
After I find the root causes, I will merge the PR firstly, then I will fix the test failure in the next PR. :)